source/SH3/graphics/texture.cpp

```diff
@@ -4,44 +4,62 @@
  *
  *  @copyright 2016  Palm Studios
  *
- *
  *  @date 2-1-2017
  *
  *  @author Jesse Buhagiar
  */
 #include <SH3/graphics/texture.hpp>
 #include <SH3/system/log.hpp>
+#include <SH3/arc/vfile.hpp>
+
+#include <cassert>
+
 
 using namespace sh3_graphics;
 
-std::string texture::load_error::message() const
+//TODO: Scale the texture and then
+int sh3_texture::Load(sh3_arc& mft, const std::string& filename)
 {
-    std::string error;
-    switch(result)
-    {
-    case load_result::SUCCESS:
-        error = "Success";
-        break;
-    case load_result::INVALID_CHUNK:
-        error = "Invalid chunk";
-        break;
-    case load_result::END_OF_FILE:
-        error = "End of file";
-        break;
-    }
-    return error;
-};
-
-texture::load_error sh3_graphics::texture::Load(const std::string& filename, sh3_arc& arc)
-{
-    load_error error;
+    sh3_texture_header header;
+    sh3_arc_vfile file(mft, filename);
+    sh3_arc_vfile::read_error e;
+
+    int ret = file.ReadData(&header, sizeof(sh3_texture_header), e); // Read in the header
+
+    if(ret != sizeof(sh3_texture_header))
+        Log(LogLevel::WARN, "sh3_texture::Load( ): Header only read %d bytes of data! It should have been %d", ret, sizeof(sh3_texture_header));
```

Comment:
We may want to fallback on a dummy texture when this happens, since it would be very cumbersome to ensure we don't read uninitialized struct fields.
For now this would be a good place to `die()`, though add a `//TODO` that we want to use a fallback texture in the future.